ttnn/cpp/ttnn/operations/reduction/topk/device/kernels/compute/topk_final.cpp

```diff
@@ -0,0 +1,166 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstdint>
+
+#include "compute_kernel_api/eltwise_unary/eltwise_unary.h"
+#include "compute_kernel_api.h"
+#include "compute_kernel_api/transpose_wh.h"
+#include "compute_kernel_api/tile_move_copy.h"
+#include "compute_kernel_api/unpack.h"
+#include "compute_kernel_api/pack.h"
+
+// topk llk needs a global variable atm
+// this can only be removed once that's fixed
+int32_t topk_replay_init = 0;
+
+namespace NAMESPACE {
+void MAIN {
+    constexpr uint32_t input_cb_index = get_compile_time_arg_val(0);
+    constexpr uint32_t index_cb_index = get_compile_time_arg_val(1);
+    constexpr uint32_t input_transposed_cb_index = get_compile_time_arg_val(2);
+    constexpr uint32_t index_transposed_cb_index = get_compile_time_arg_val(3);
+    constexpr uint32_t values_cb_index = get_compile_time_arg_val(4);
+    constexpr uint32_t output_ind_cb_index = get_compile_time_arg_val(5);
+    constexpr uint32_t Ht = get_compile_time_arg_val(6);
+    constexpr uint32_t Wt = get_compile_time_arg_val(7); // Leftover row size after multicore processing
+    constexpr uint32_t K = get_compile_time_arg_val(8);
+    constexpr uint32_t logk = get_compile_time_arg_val(9);
+    constexpr uint32_t logWt = get_compile_time_arg_val(10);
+
+    // dest indices for where to unpack the tiles for the llk
+    // the input goes in index 0,1 and the index goes in index 2,3
+    constexpr uint32_t input_dest_start = 0;
+    constexpr uint32_t index_dest_start = 2;
+    constexpr uint32_t input_dest_end = 1;
+    constexpr uint32_t index_dest_end = 3;
+    // init pack, compute and unpack
+
+
+    init_sfpu(input_cb_index);
+    ckernel::topk_tile_init();
+    bool ascending = false;
```

Comment:
Is this something that will be uplifted later? Make it `constexpr`.

Reply:
old variable that needs to be deleted